src/m365/base/TeamsCallReport.ts

```diff
@@ -0,0 +1,91 @@
+import { Logger } from '../../cli';
+import {
+  CommandOption
+} from '../../Command';
+import GlobalOptions from '../../GlobalOptions';
+import request from '../../request';
+import Utils from '../../Utils';
+import GraphCommand from "./GraphCommand";
+
+interface CommandArgs {
+  options: DateTimeOptions;
+}
+
+interface DateTimeOptions extends GlobalOptions {
+  fromDateTime: string;
+  toDateTime?: string;
+}
+
+export default abstract class TeamsCallReport extends GraphCommand {
+  public abstract get usageEndpoint(): string;
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const endpoint: string = `${this.resource}/beta/communications/callRecords/${this.usageEndpoint}(fromDateTime=${encodeURIComponent(args.options.fromDateTime)},toDateTime=${encodeURIComponent(args.options.toDateTime!)})`;
```

Comment:
Is the `!` meant to assume that the value is specified? What's going to happen if it's not?

Reply:
Hi,

The parameter `toDateTime` is optional as requested in #1771, but to run the API call it is mandatory.

To enforce this, a value is checked for during the `validate` method and assign it a string of *now* if not:
```
    if (!args.options.toDateTime) {
      args.options.toDateTime = new Date().toISOString();
    }
```

The reason the check is done in `validate` rather than in `commandAction` is we are also comparing the date period between fromDateTime and toDateTime doesn't exceed 90 days (see #1771):
```
    if (Math.ceil((new Date(args.options.toDateTime).getTime() - new Date(args.options.fromDateTime).getTime()) / (1000 * 3600 * 24)) > 90) {
      return 'The maximum number of days between fromDateTime and toDateTime cannot exceed 90'
    }
```

However, `encodeURIComponent` in `commandAction` is expecting `string` but as `toDateTime` is optional, it is not guaranteed to come in as an argument from the user, but I believed it *should* as it runs through `validate` first, hence the bang.

I suppose in `validate` I could use a temporary value in lieu of `toDateTime` to do the date period check. Then assign another value to `toDateTime` (if not specified) before the line in question in `commandAction`. I thought it was cleaner to do the check once rather than twice, though.

I'll leave it up to you :+1:

Reply:
Thanks for clarification 👍